packaging/standalone/src/main/distribution/shell-scripts/bin/Neo4j-Management/Get-Neo4jWindowsServiceName.ps1

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) 2002-2015 "Neo Technology,"
+# Network Engine for Objects in Lund AB [http://neotechnology.com]
+#
+# This file is part of Neo4j.
+#
+# Neo4j is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+<#
+.SYNOPSIS
+Retrieves the name of the Windows Service from the configuration information
+
+.DESCRIPTION
+Retrieves the name of the Windows Service from the configuration information
+
+.PARAMETER Neo4jServer
+An object representing a valid Neo4j Server object
+
+.EXAMPLE
+'C:\Neo4j' | Get-Neo4jServer | Get-Neo4jWindowsServiceName 
+
+Retrieves the name of the Windows Service for the Neo4j Database at 'C:\Neo4j'
+
+.OUTPUTS
+System.String
+The name of the Windows Service or $null if it could not be determined
+
+.NOTES
+This function is private to the powershell module
+
+#>
+Function Get-Neo4jWindowsServiceName
```

Comment:
Either way, there's an issue.  I prefer to take the tact of only querying for config settings at the last possible point you need it.  Otherwise there is a chance a user (or another process) can modify the config file after between the time you read it and action against it.  It's more an issue in the older powershell scripts.